spec.bs

```diff
@@ -1529,19 +1546,31 @@ Several APIs specific to fenced frames are defined on the {{Fence}} interface.
   
   1. If |instance|'s [=fenced frame config instance/is ad component=] is true, then return.
 
-  1. If the [=relevant settings object=]'s [=environment settings object/origin=] and |instance|'s
-     [=fenced frame config instance/mapped url=]'s [=url/origin=] are not [=same origin=], then
-     return.
-
   1. If |instance|'s [=fenced frame config instance/fenced frame reporter=] is null, then return.
 
-  1. If |event| is a {{DOMString}}, run [=report a private aggregation event=] using |instance|'s
-     [=fenced frame config instance/fenced frame reporter=] with |event|.
+  1. If |event| is a {{DOMString}}:
+
+     1. If the [=relevant settings object=]'s [=environment settings object/origin=] and
+        |instance|'s [=fenced frame config instance/mapped url=]'s [=url/origin=] are not [=same
+        origin=], then return.
```

Comment:
Just to be clear, we don't need to consider the other conditions that we do on L1566, right? About the `crossOriginExposed` bit being true and so on, to allow this report to go through even in the cross-origin case?

Reply:
Correct. In this path, `event` is a `DOMString`, not a `FenceEvent`, so there's no `crossOriginExposed` bit that can be checked.

Reply:
Oh yeah, of course. But do we also not need to concern ourself with `|instance|'s [=fenced frame config instance/cross-origin reporting allowed=]`?

Reply:
No. `DOMString` events don't support cross-origin beacons.

The only way for this feature to be safely supported is if we can get opt-in from both the document loaded from the fenced frame config as well as the document sending the `reportEvent()` beacon. The former is through the response header whose value is noted in `|instance|'s [=fenced frame config instance/cross-origin reporting allowed=]`. The latter is through the `crossOriginExposed` bit.

Since a `DOMString` event is just that, a `DOMString`, it has no `crossOriginExposed` bit, i.e. no way to opt in for cross-origin beacons. Because of that, there's no way we can safely support this feature for DOMString events. So it doesn't matter what kind of opt-in the document loaded with the fenced frame config had; it will just be ignored.

The only place we need to check `|instance|'s [=fenced frame config instance/cross-origin reporting allowed=]?` this is in the "If event is a [FenceEvent](https://github.com/WICG/fenced-frame/pull/152#dictdef-fenceevent):" section.

Reply:
Got it, thanks.